### PR TITLE
Upgrade partials to manifolds

### PR DESCRIPTION
**Pull request by Affie**
- rm getManifolds, fix compare conflict and update cov
- Range2D
- Bearing2D
- PartialPriorYawPose2

Broken Testset for RoME summary: 
  Pass:325
  Broken:4
  Fail:32
  Error:28

needs https://github.com/JuliaRobotics/IncrementalInference.jl/pull/1356

### CODE REVIEW

File: src/factors/Range2D.jl [hunk 1]
```diff
@@ -9,7 +9,7 @@ mutable struct Point2Point2Range{D <: IIF.SamplableBelief} <: IncrementalInferen
 end
 
 function getSample(cfo::CalcFactor{<:Point2Point2Range}, N::Int=1)
-  return (reshape(rand(cfo.factor.Z,N),1,N),)
+  return (rand(cfo.factor.Z, N),)
```
Comment (dehann) [inline src/factors/Range2D.jl:12]: vector of tangents?

File: src/RoME.jl [hunk 2]
```diff
@@ -32,13 +32,12 @@ using Manifolds: ProductGroup, ProductManifold, SpecialEuclidean, ProductRepr, S
 
 
 import Base: +, \, convert
-import TransformUtils: ⊖, ⊕, convert, compare, ominus, veeQuaternion
```
Comment (dehann) [inline src/RoME.jl:35]: oh, was that the problem, thanks!

File: test/testBearing2D.jl [hunk 3]
```diff
@@ -2,6 +2,47 @@
...
+f = Pose2Point2Bearing(Normal(pi/4,0.05))
+
+for (i,p) in enumerate(ps)
+    # @show p
+    r = calcFactorResidualTemporary(f, (Pose2, Point2), m, (p, q))
+    # @warn "comp: $(isapprox(r, rs[i], atol=0.001))" a=r b=rs[i]
+    @test isapprox(r, rs[i], atol=0.001) 
+end
+
+end
```
Comment (dehann) [inline test/testBearing2D.jl:46]:
```suggestion
@warn "Bearing2D, must still test factor gradients, which will also verify the sign on residual function calculations"
@test_broken false
```

File: src/factors/Range2D.jl (hunk 1 above)
Comment (Affie) [reply src/factors/Range2D.jl:12]: It doesn't seem to matter, will keep like this as a test for now as it should not matter as samples are a Vector{P} and P can be a Float

**Comment by Affie** [comment]
PR is getting big so will do 3d partials seperately

**Pull request merged**